src/jit/lower.cpp

```diff
@@ -491,7 +491,63 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
         }
         break;
     case GT_NEG:
-        NYI("GT_NEG of TYP_LONG");
+        {
+            GenTree* op1 = tree->gtGetOp1();
+            noway_assert(op1->OperGet() == GT_LONG);
+
+            // TODO-Cleanup: Identical code exists in DecomposeStoreInd,
+            // it should be moved to a function similar to CreateTemporary
+            GenTreeStmt* newStmt = comp->fgInsertEmbeddedFormTemp(&op1->gtOp.gtOp1);
+            if (newStmt->gtStmtIsTopLevel())
+            {
+                for (GenTreeStmt* nextEmbeddedStmt = newStmt->gtStmtNextIfEmbedded();
+                     nextEmbeddedStmt != nullptr;
+                     nextEmbeddedStmt = nextEmbeddedStmt->gtStmt.gtStmtNextIfEmbedded())
+                {
+                    comp->compCurStmt = nextEmbeddedStmt;
+                    comp->fgWalkTreePost(&nextEmbeddedStmt->gtStmt.gtStmtExpr, Lowering::DecompNodeHelper, this, true);
+                }
+
+                comp->compCurStmt = curStmt;
+            }
+
+            newStmt = comp->fgInsertEmbeddedFormTemp(&op1->gtOp.gtOp2);
+            assert(!newStmt->gtStmtIsTopLevel());  
```

Comment:
I'm wondering why this case of `fgInsertEmbeddedFormTemp` doesn't also need the logic from above to recurse through the possible embedded statements in case a new top-level stmt is created. Is there some reason the case handled there won't occur in this case? I agree with the "TODO-Cleanup" note!